sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -747,4 +776,182 @@ String getParentLink() {
+            context, spanName, database.getId(), database.getClient().getServiceEndpoint());
+    }
+
+    private Mono<CosmosAsyncItemResponse<Object>> deleteItemInternal(
+        String itemId,
+        RequestOptions requestOptions) {
+        return this.getDatabase()
+            .getDocClientWrapper()
+            .deleteDocument(getItemLink(itemId), requestOptions)
+            .map(response -> ModelBridgeInternal.createCosmosAsyncItemResponseWithObjectType(response))
+            .single();
+    }
+
+    private <T> Mono<CosmosAsyncItemResponse<T>> replaceItemInternal(
+        Class<T> itemType,
+        String itemId,
+        Document doc,
+        CosmosItemRequestOptions options,
+        Context context) {
+        String spanName = "replaceItem." + this.getId();
+        Mono<CosmosAsyncItemResponse<T>> responseMono = replaceItemInternal(itemType, itemId, doc, options);
+        return database.getClient().getTracerProvider().traceEnabledCosmosItemResponsePublisher(responseMono,
+            context, spanName, database.getId(), database.getClient().getServiceEndpoint());
+    }
+
+    private <T> Mono<CosmosAsyncItemResponse<T>> replaceItemInternal(
+        Class<T> itemType,
+        String itemId,
+        Document doc,
+        CosmosItemRequestOptions options) {
+       return this.getDatabase()
+            .getDocClientWrapper()
+            .replaceDocument(getItemLink(itemId), doc, ModelBridgeInternal.toRequestOptions(options))
+            .map(response -> ModelBridgeInternal.createCosmosAsyncItemResponse(response, itemType))
+            .single();
+    }
+
+    private <T> Mono<CosmosAsyncItemResponse<T>> upsertItemInternal(T item, CosmosItemRequestOptions options, Context context) {
+        String spanName = "upsertItem." + this.getId();
+        Mono<CosmosAsyncItemResponse<T>> responseMono = upsertItemInternal(item, options);
+        return database.getClient().getTracerProvider().traceEnabledCosmosItemResponsePublisher(responseMono,
+            context, spanName, database.getId(), database.getClient().getServiceEndpoint());
+    }
+
+    private <T> Mono<CosmosAsyncItemResponse<T>> upsertItemInternal(T item, CosmosItemRequestOptions options) {
+        @SuppressWarnings("unchecked")
+        Class<T> itemType = (Class<T>) item.getClass();
+        return this.getDatabase().getDocClientWrapper()
+            .upsertDocument(this.getLink(), item,
+                ModelBridgeInternal.toRequestOptions(options),
+                true)
+            .map(response -> ModelBridgeInternal.createCosmosAsyncItemResponse(response, itemType))
+            .single();
+    }
+
+    private <T> Mono<CosmosAsyncItemResponse<T>> readItemInternal(
+        String itemId, PartitionKey partitionKey,
+        RequestOptions requestOptions, Class<T> itemType,
+        Context context) {
+        String spanName = "readItem." + this.getId();
```

Comment:
`"readItem." + this.getId()` can be pre-computed in container rather than re-computing each time.

please pre-compute in the container.

and for similar other operations.